core/server/src/test/java/alluxio/worker/file/UnderFileSystemManagerTest.java

```diff
@@ -315,12 +316,56 @@ public void getInputStreamAtPositionTest() throws Exception {
     UnderFileSystemManager manager = new UnderFileSystemManager();
     long id = manager.openFile(SESSION_ID, new AlluxioURI(uniqPath));
     InputStream in = manager.getInputStreamAtPosition(id, position);
-    Assert.assertEquals(mMockInputStream, in);
+    Assert.assertEquals(mMockInputStream, Whitebox.getInternalState(in, "in"));
     Mockito.verify(mMockInputStream).skip(position);
     in.close();
   }
 
   /**
+   * Tests getting an input stream to a valid file at a position returns the correct input
+   * stream, then reading and getting the next position returns the same stream.
+   */
+  @Test
+  public void getInputStreamAtPositionCacheTest() throws Exception {
+    String uniqPath = PathUtils.uniqPath();
+    long position = 0;
+    Mockito.when(mMockUfs.exists(uniqPath)).thenReturn(true);
+    UnderFileSystemManager manager = new UnderFileSystemManager();
+    long id = manager.openFile(SESSION_ID, new AlluxioURI(uniqPath));
+    InputStream in = manager.getInputStreamAtPosition(id, position);
+    Assert.assertEquals(mMockInputStream, Whitebox.getInternalState(in, "in"));
+    long nextPosition = 100;
+    Mockito.when(mMockInputStream.skip(nextPosition)).thenReturn(nextPosition);
+    in.skip(nextPosition - position);
+    InputStream in2 = manager.getInputStreamAtPosition(id, nextPosition);
+    Assert.assertEquals(in, in2);
+    Mockito.verify(mMockInputStream, Mockito.never()).skip(position);
+    in.close();
+  }
+
+  /**
+   * Tests getting an input stream to a valid file at a position returns the correct input
```

Comment:
I refactored the unit test to improve the style, but I don't think we can combine the two tests. I also updated the comment to sound less overlapping.